gossip-protocol-spec.md

```diff
@@ -1064,5 +1081,65 @@ struct RestartHeaviestFork {
 ```
 </details>
 
+# Addendum
+
+## IP Echo Server
+
+The IP Echo Server is a server running on the TCP socket on the same gossip address. (e.g. if the node is running the gossip service on UDP socket 192.0.0.1:9000, then the IP server port is running on the same TCP socket 192.0.0.1:9000).
+
+Before the node starts the gossip service, the node first needs to:
+- find out the shred version of the cluster,
+- discover its public IP address,
+- check TCP/UDP port reachability.
+
+All those are checked via the IP echo server (which is running on every validator) on one of the entrypoint nodes that the node was provided with.
```

Comment:
> All those are checked via the IP echo server (which is running on every validator) on one of the entrypoint nodes that the node was provided with.

All of these are discovered via the IP echo server running on one of the provided entrypoint nodes. Note: all validators run an IP Echo Server.